File: tests/analyze_dom.py
import os
import sys
from bs4 import BeautifulSoup

def analyze_file(file_path):
    print(f"\n--- Analyzing {os.path.basename(file_path)} ---")
    if not os.path.exists(file_path):
        print("File not found")
        return

    with open(file_path, 'r', encoding='utf-8') as f:
        html = f.read()
    
    soup = BeautifulSoup(html, 'html.parser')

    # 1. Check for Reviews tab
    print("\n[Search: Reviews Tab]")
    tabs = soup.find_all('button', role='tab')
    print(f"Found {len(tabs)} tabs with role='tab'")
    for i, tab in enumerate(tabs):
        print(f"  Tab {i}: text='{tab.get_text(strip=True)}' aria-label='{tab.get('aria-label')}' class='{tab.get('class')}'")

    # Search for specific text 'Reviews'
    print("\n[Search: Text 'Reviews']")
    reviews_els = soup.find_all(lambda tag: tag.name in ['button', 'div', 'span'] and 'Reviews' in tag.get_text())
    for i, el in enumerate(reviews_els[:10]):
        print(f"  Match {i}: <{el.name} class='{el.get('class')}' aria-label='{el.get('aria-label')}' role='{el.get('role')}'>{el.get_text(strip=True)[:50]}...")

    # 2. Check for Sort button
    print("\n[Search: Sort Button]")
    sort_btns = soup.find_all('button', attrs={'aria-label': lambda x: x and 'Sort' in x})
    print(f"Found {len(sort_btns)} buttons with aria-label containing 'Sort'")
    for i, btn in enumerate(sort_btns):
         print(f"  Btn {i}: aria-label='{btn.get('aria-label')}' class='{btn.get('class')}'")

step1_path = os.path.join(os.getcwd(), 'tests/output/debug/step1_details_overview.html')
step3_path = os.path.join(os.getcwd(), 'tests/output/debug/step3_sort_menu.html')

analyze_file(step1_path)
analyze_file(step3_path)

File: tests/analyze_dom.js
const fs = require('fs');
const path = require('path');
const { parse } = require('node-html-parser');

const step1Path = path.join(__dirname, 'output/debug/step1_details_overview.html');
const step3Path = path.join(__dirname, 'output/debug/step3_sort_menu.html');

function analyzeFile(filePath) {
    console.log(`\n--- Analyzing ${path.basename(filePath)} ---`);
    if (!fs.existsSync(filePath)) {
        console.log('File not found');
        return;
    }
    const html = fs.readFileSync(filePath, 'utf8');
    const root = parse(html);

    // 1. Check for details panel role
    const mains = root.querySelectorAll('div[role="main"]');
    console.log(`Found ${mains.length} div[role="main"]`);
    mains.forEach((main, i) => {
        const hasFeed = main.querySelector('div[role="feed"]');
        console.log(`  Main ${i}: hasFeed=${!!hasFeed} class="${main.getAttribute('class')}" aria-label="${main.getAttribute('aria-label')}"`);
    });

    // 2. Check for Reviews tab
    // Try broadly
    const tabs = root.querySelectorAll('button[role="tab"]');
    console.log(`Found ${tabs.length} tabs`);
    tabs.forEach((tab, i) => {
        console.log(`  Tab ${i}: text="${tab.textContent.trim()}" aria-label="${tab.getAttribute('aria-label')}" class="${tab.getAttribute('class')}"`);
    });

    // 3. Search for "Reviews" text in any button/div
    console.log('Searching for elements with text "Reviews"...');
    const reviewsEls = root.querySelectorAll('*');
    let reviewsCount = 0;
    for (const el of reviewsEls) {
        if (reviewsCount > 5) break;
        const text = el.textContent || '';
        const aria = el.getAttribute('aria-label') || '';
        if ((text.includes('Reviews') || aria.includes('Reviews')) && (el.tagName === 'BUTTON' || el.tagName === 'DIV')) {
            // filter out huge containers
            if (text.length < 100) {
                console.log(`  Match: <${el.tagName} class="${el.getAttribute('class')}" aria-label="${aria}" role="${el.getAttribute('role')}">${text}</${el.tagName}>`);
                reviewsCount++;
            }
        }
    }

    // 4. Check for Sort button
    console.log('Searching for "Sort"...');
    // ... similar logic
}

analyzeFile(step1Path);
analyzeFile(step3Path);
